lua/wikis/commons/TournamentsListing/CardList/Custom.lua

```diff
@@ -8,20 +8,61 @@
 local Lua = require('Module:Lua')
 
 local Arguments = Lua.import('Module:Arguments')
+local Array = Lua.import('Module:Array')
 local Class = Lua.import('Module:Class')
+local Info = Lua.import('Module:Info')
+local Logic = Lua.import('Module:Logic')
+local Table = Lua.import('Module:Table')
 
 local TournamentsListing = Lua.import('Module:TournamentsListing/CardList')
 
+local HtmlWidgets = Lua.import('Module:Widget/Html/All')
+
+local DEFAULT_START_YEAR = Info.startYear
+local DEFAULT_END_YEAR = tonumber(os.date('%Y')) --[[@as integer]]
+
 local CustomTournamentsListing = Class.new()
 
 ---@param frame Frame
----@return Html?
+---@return Html|Widget?
 function CustomTournamentsListing.run(frame)
-	local tournamentsListing = TournamentsListing(Arguments.getArgs(frame))
+	local args = Arguments.getArgs(frame)
+
+	if Logic.readBool(args.byYear) then
+		args.byYear = nil
+		return CustomTournamentsListing.byYear(args)
+	end
+
+	return TournamentsListing(args):create():build()
+end
+
+---@param args table
+---@return Widget?
+function CustomTournamentsListing.byYear(args)
```

Comment:
inb4 Rathoz is going to say that this should be a separate widget invoking this Custom.run, which arguably makes sense. Otherwise, any wiki with a custom would need to duplicate this logic.